extension/extension.go

```diff
@@ -48,7 +47,7 @@ type ConfigWatcher interface {
 	// The extension owns the `confmap.Conf`. Callers must ensure that it's safe for
 	// extensions to store the `conf` pointer and use it concurrently with any other
 	// instances of `conf`.
-	NotifyConfig(ctx context.Context, conf *confmap.Conf) error
+	NotifyConfig(ctx context.Context, data any) error
```

Comment:
Please avoid "any" in public APIs as much as possible. This is usually a sign of a poorly designed API.

Reply:
Agreed, this was mostly just done as a PoC. We agreed to use `confmap.Conf` during this week's SIG meeting (which I'd prefer to do anyway), so I'll be reverting this.